Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm

```diff
@@ -216,10 +230,22 @@ bool experimentalFeatureEnabled(const String& key, bool defaultValue)
     parameters.networkSessionParameters.enablePrivateClickMeasurementDebugMode = experimentalFeatureEnabled(WebPreferencesKey::privateClickMeasurementDebugModeEnabledKey());
 }
 
-bool WebsiteDataStore::useNetworkLoader()
+std::optional<bool> WebsiteDataStore::useNetworkLoader()
 {
 #if HAVE(NETWORK_LOADER)
-    return experimentalFeatureEnabled(WebPreferencesKey::cFNetworkNetworkLoaderEnabledKey());
+    auto enabledState = experimentalFeatureEnabled(WebPreferencesKey::cFNetworkNetworkLoaderEnabledKey());
+    auto enabledInternalState = optionalExperimentalFeatureEnabled(WebPreferencesKey::cFNetworkNetworkLoaderInternalStateEnabledKey(), std::nullopt, "InternalDebug"_s);
+    if (!enabledState && !enabledInternalState)
+        return std::nullopt;
+
+    if (enabledState && (!enabledInternalState || !*enabledInternalState)) {
+        enabledInternalState = true;
+        setExperimentalFeatureState(WebPreferencesKey::cFNetworkNetworkLoaderInternalStateEnabledKey(), *enabledInternalState, "InternalDebug"_s);
```

Comment:
I don't think we want to be setting state in this function.  I think instead, we should make it so that Safari sets true, false, or unset at least for this particular feature.